### PR TITLE
Support switching tabstrip orientation 

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/25472

We can now switch the tabstrip orientation from horizontal to vertical
and vice versa at runtime.

https://user-images.githubusercontent.com/5474642/198002806-5d95ae36-8dbb-4898-8931-d64ed4c2f98a.mov

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Automated:
VerticalTabStripBrowserTest.Toggle

### Manual
* Open context menu for tab
* Click "Use vertical tabs"
* Should be able to switch the orientation in turn.

### CODE REVIEW

**Review by fallaciousreasoning — APPROVED** [review]
LGTM! This looks awesome @sangwoo108!

File: browser/ui/views/frame/vertical_tab_strip_widget_delegate_view.cc [hunk 1]
```diff
@@ -112,7 +112,9 @@ void VerticalTabStripWidgetDelegateView::UpdateWidgetBounds() {
     return;
 
   auto* widget = GetWidget();
-  DCHECK(widget);
+  if (!widget)
```
Comment (simonhong) [inline browser/ui/views/frame/vertical_tab_strip_widget_delegate_view.cc:115]: Just curious what makes this change?

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: browser/ui/views/frame/vertical_tab_strip_widget_delegate_view.cc (hunk 1 above)
Comment (sangwoo108) [reply browser/ui/views/frame/vertical_tab_strip_widget_delegate_view.cc:115]: Yeah, it starts to fail to pass the DCHECK on the startup.

File: browser/ui/views/tabs/brave_tab_context_menu_contents.h [hunk 2]
```diff
@@ -52,6 +53,9 @@ class BraveTabContextMenuContents : public ui::SimpleMenuModel::Delegate {
   void ExecuteBraveCommand(int command_id);
   bool IsBraveCommandId(int command_id) const;
 
+  const PrefService* GetPrefs() const;
+  PrefService* GetPrefs();
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply browser/ui/views/tabs/brave_tab_context_menu_contents.h]: THanks! Removed.

File: chromium_src/chrome/browser/ui/views/tabs/tab_group_views.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_VIEWS_TABS_TAB_GROUP_VIEWS_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_VIEWS_TABS_TAB_GROUP_VIEWS_H_
+
+#define OnGroupVisualsChanged                   \
+  OnGroupVisualsChanged_Unused() {}             \
+  const TabSlotController& controller() const { \
```
Comment (goodov) [inline chromium_src/chrome/browser/ui/views/tabs/tab_group_views.h:11]: `tab_slot_controller()` or just put `GetBrowser()` here.

File: chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc [hunk 4]
```diff
@@ -38,7 +39,13 @@ void TabDragController::Init(TabDragContext* source_context,
   TabDragControllerChromium::Init(source_context, source_view, dragging_views,
                                   mouse_offset, source_view_offset,
                                   initial_selection_model, event_source);
-  if (!tabs::features::ShouldShowVerticalTabs())
+  auto* widget = source_view->GetWidget()->GetTopLevelWidget();
+  DCHECK(widget);
+  const auto* browser =
+      static_cast<BraveBrowserFrame*>(widget)->browser_view()->browser();
```
Comment (goodov) [inline chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc:45]: can we remove this cast? I see upstream uses something like this:
```
  BrowserView::GetBrowserViewForNativeWindow(
      GetAttachedBrowserWidget()->GetNativeWindow())
      ->browser();
```

will this work?
```
  BrowserView::GetBrowserViewForNativeWindow(
      source_context->GetWidget()->GetNativeWindow())
      ->browser();
```

Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc:45]: Yeah, that would work. I was on the fence cuz GetBrowserViewNativeWindow() should be avoided if possible. But that'd be more safe way to get browser.

File: chromium_src/chrome/browser/ui/views/tabs/tab_group_views.h (hunk 3 above)
Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/tabs/tab_group_views.h:11]: Cool. I love the latter.

**Review by goodov — APPROVED** [review]

**Pull request merged**